js/parser.js

```diff
@@ -191,11 +196,181 @@ var ERMrest = (function(module) {
                     }
                 }
 
-                context.filter = new ParsedFilter(type);
-                context.filter.setFilters(filters);
+                this._firstFilter = new ParsedFilter(type);
+                this._firstFilter.setFilters(filters);
+            }
+        }
+
+    }
+
+    Location.prototype = {
```

Comment:
The getters in this prototype should be documented (jsdoc comments). Most likely in the following style so they appear as properties.

```javascript
/**
 * Some descrpitoin
 * @type {String}
 */
get xyz() {
...
}
```